sqlglot/optimizer/qualify_columns.py

```diff
@@ -527,12 +532,21 @@ def _convert_columns_to_dots(scope: Scope, resolver: Resolver) -> None:
         scope.clear_cache()
 
 
-def _qualify_columns(scope: Scope, resolver: Resolver, allow_partial_qualification: bool) -> None:
+def _qualify_columns(
+    scope: Scope,
+    resolver: Resolver,
+    allow_partial_qualification: bool,
+    pseudocolumns: t.Optional[t.Set[str]] = None,
+) -> None:
     """Disambiguate columns, ensuring each column specifies a source"""
     for column in scope.columns:
         column_table = column.table
         column_name = column.name
 
+        if pseudocolumns and not column.table and column_name.upper() in pseudocolumns:
+            column.replace(exp.Pseudocolumn(**column.args))
+            continue
```

Comment:
Can we refactor `_expand_stars` now? What about the `LEVEL` pseudocolumn logic at the start of `qualify_columns`?

Reply:
Sounds good 👍